examples/knative/go/kn-go-echo/server.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+func main() {
+	ctx := context.Background()
+
+	client, err := cloudevents.NewClientHTTP()
+	if err != nil {
+		log.Fatalf("creating HTTP client, %v", err)
+	}
+
+	log.Println("listening on :8080")
+	if err := client.StartReceiver(ctx, receive); err != nil {
+		log.Fatalf("start receiver: %s", err.Error())
+	}
+}
+
+func receive(ctx context.Context, event cloudevents.Event) *cloudevents.Event {
+	fmt.Printf("***cloud event***\n%s", event)
+
+	response := event
```

Comment:
Please use https://pkg.go.dev/github.com/cloudevents/sdk-go/v2@v2.4.1/event#Event.Clone

Reply:
I don't even know why I made a copy. It's not like the event being passed in is a pointer. It's just a copy of the original event being sent in. When I send a pointer, it's a pointer to a copy of the event anyway.

Reply:
I think it's ok to pass along the original event, even if it has some reference types. Let's see what you think with a new commit.